src/tree/browser/components/treetable-navigator.tsx

```diff
@@ -0,0 +1,169 @@
+/********************************************************************************
+ * Copyright (C) 2024 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License as outlined in the LICENSE File
+ ********************************************************************************/
+
+import { CDTTreeItemResource, CDTTreeItem } from '../../common';
+
+export interface TreeNavigatorProps<T extends CDTTreeItemResource> {
+    ref: React.RefObject<HTMLDivElement>;
+    rowIndex: Map<string, number>;
+    expandedRowKeys: string[];
+    expand: (expanded: boolean, record: CDTTreeItem<T>) => void;
+    select: (record: CDTTreeItem<T>) => void;
+}
+
+/**
+ * TreeNavigator is a helper class to navigate
+ * through a tree table.
+ */
+export class TreeNavigator<T extends CDTTreeItemResource = CDTTreeItemResource> {
+    constructor(private readonly props: TreeNavigatorProps<T>) {}
+
+    next(node: CDTTreeItem) {
```

Comment:
Nit: probably more of a personal taste, but I think explicitly writing `public` makes the code easier readable. Helps me to spot members quicker.

Reply:
Sounds good to me, I will add an eslint rule to enfore it.

https://typescript-eslint.io/rules/explicit-member-accessibility/